pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1436,6 +1443,18 @@ static __always_inline u64 get_ctime_nanosec_from_dentry(struct dentry *dentry)
     return get_ctime_nanosec_from_inode(d_inode);
 }
 
+static __always_inline file_info_t get_file_info(struct file *file)
+{
+    file_info_t file_info = {};
+    if (file != NULL) {
+        file_info.pathname_p = get_path_str(GET_FIELD_ADDR(file->f_path));
+        file_info.ctime = get_ctime_nanosec_from_file(file);
+        file_info.device = get_dev_from_file(file);
+        file_info.inode = get_inode_nr_from_file(file);
+    }
+    return file_info;
```

Comment:
So correct me if I am wrong, but I don't think there is a problem with it anyways.
I think the compiler knows how to save it in a way that is accessible after returning from the function - as evident it is valid to return struct from a function (and not a pointer).
I guess that behind the scenes it just saves memory for it on the stack of the calling function and dereference it from the inner function (after compilation) or pass pointer.

Reply:
I understand that in this particular case it may be ok, but I prefer not to make such assumptions.
As a best practice, can you define file_info out of this function and pass it as an argument?

Reply:
Ok, I understand what your'e saying about returning a struct and not a pointer.
It seems to be a valid case in C (https://stackoverflow.com/questions/9653072/return-a-struct-from-a-function-in-c)

Reply:
OK, so I returned it to be as it was